docs/encyclopedia/retry-policies.mdx

```diff
@@ -101,13 +103,151 @@ Non-Retryable Errors = []
 - **Use case:** Use this attribute to ensure that retries do not continue indefinitely.
   In most cases, we recommend using the Workflow Execution Timeout for [Workflows](/workflows) or the Schedule-To-Close Timeout for Activities to limit the total duration of retries, rather than using this attribute.
 
-### Non-Retryable Errors
+### Non-Retryable Errors {#non-retryable-errors}
 
-- **Description:** Specifies errors that shouldn't be retried.
-  - **Default is none.**
-  - Errors are matched against the `type` field of the [Application Failure](/references/failures#application-failure).
-  - If one of those errors occurs, a retry does not occur.
-- **Use case:** If you know of errors that should not trigger a retry, you can specify that, if they occur, the execution is not retried.
+Non-Retryable Errors specify errors that shouldn't be retried.
+By default, none are specified.
+Errors are matched against the `type` field of the [Application Failure](/references/failures#application-failure).
+If one of those errors occurs, a retry does not occur.
+If you know of errors that should not trigger a retry, you can specify that, if they occur, the execution is not retried.
+
+#### Non-Retryable Errors for Activities
+
+When writing software applications, you will encounter three types of failures: transient, intermittent, and permanent.
+While transient and intermittent failures may resolve themselves upon retrying without further intervention, permanent failures will not.
+Permanent failures, by definition, require you to make some change to your logic or your input.
+Therefore, it is better to surface them than to retry them.
+
+Non-Retryable Errors are errors that will not be retried, regardless of a Retry Policy.
+
+<Tabs groupId="sdk-language" queryString>
+<TabItem value="ruby" label="Ruby">
+
+To raise a non-retryable error, specify the `non_retryable` flag when raising an `ApplicationError`:
+
+```ruby
+raise Temporalio::Error::ApplicationError.new(
+  "Invalid credit card number: #{credit_card_number}",
+  type: 'InvalidChargeAmount',
+  non_retryable: true
+)
+```
+
+This will designate the `ApplicationError` as non-retryable.
+
+</TabItem>
+<TabItem value="python" label="Python">
+
+To raise a non-retryable error, specify the `non_retryable` flag when raising an `ApplicationError`:
+
+```python
```

Comment:
The spacing looks a little off here. I think if you remove the spaces to the left, it'll line up with the code block.
<img width="903" height="268" alt="Screenshot 2025-12-04 at 15 32 09" src="https://github.com/user-attachments/assets/98b99d97-f11d-4fd3-8d05-72cf3fe78770" />